miden/tests/integration/stdlib/math/secp256k1_scalar_field.rs

```diff
@@ -0,0 +1,342 @@
+use super::{build_test, Felt};
+use std::cmp::PartialEq;
+use std::ops::Mul;
+
+/// Secp256k1 scalar field element, kept in Montgomery form
+#[derive(Copy, Clone, Debug)]
+struct ScalarField {
+    limbs: [u32; 8],
+}
+
+impl ScalarField {
+    fn zero() -> Self {
+        Self { limbs: [0u32; 8] }
+    }
+
+    fn one() -> Self {
+        Self {
+            limbs: [801750719, 1076732275, 1354194884, 1162945305, 1, 0, 0, 0],
+        }
+    }
+
+    /// See https://github.com/itzmeanjan/secp256k1/blob/6e5e654823a073add7d62b21ed88e9de9bb06869/field/scalar_field_utils.py#L41-L46
+    fn mac(a: u32, b: u32, c: u32, carry: u32) -> (u32, u32) {
+        let tmp = a as u64 + (b as u64 * c as u64) + carry as u64;
```

Comment:
I tested and its ok as it is

```rust
pub fn mac(a: u32, b: u32, c: u32, carry: u32) -> (u32, u32) {
    let (a, b, c, carry) = (a as u64, b as u64, c as u64, carry as u64);

    // these will never overflow because they are originally `u32`.
    let bc = b * c;
    let acarry = a + carry;

    let result = bc
        .checked_add(acarry)
        .expect("this was the potential problem");

    let hi = (result >> 32) as u32;
    let lo = result as u32;
    (hi, lo)
}

#[test]
fn mac_overflow() {
    let (hi, lo) = mac(u32::MAX, u32::MAX, u32::MAX, u32::MAX);
    assert_ne!(0, hi);
    assert_ne!(0, lo);
}
```

Reply:
Thanks @vlopes11